$OpenBSD: patch-sysdeps_openbsd_proctime_c,v 1.6 2013/08/26 08:43:16 ajacoutot Exp $

XXX push upstream:
Some fields have moved in struct proc from timeval to timespec.

--- sysdeps/openbsd/proctime.c.orig	Sun Jul 24 21:13:56 2011
+++ sysdeps/openbsd/proctime.c	Sat Jun  8 15:00:41 2013
@@ -71,7 +71,7 @@ calcru(struct proc *p, struct timeval *up, struct time
 {
 	quad_t totusec;
 	u_quad_t u, st, ut, it, tot;
-        long sec, usec;
+        long sec, nsec;
         struct timeval tv;
 
 	st = p->p_sticks;
@@ -85,9 +85,9 @@ calcru(struct proc *p, struct timeval *up, struct time
 	}
 
 	sec = p->p_rtime.tv_sec;
-	usec = p->p_rtime.tv_usec;
+	nsec = p->p_rtime.tv_nsec;
 
-	totusec = (quad_t)sec * 1000000 + usec;
+	totusec = (quad_t)sec * 1000000 + nsec/1000;
 
 	if (totusec < 0) {
 		/* XXX no %qd in kernel.  Truncate. */
